vkubelet/node.go

```diff
@@ -328,18 +332,20 @@ func UpdateNodeStatus(ctx context.Context, nodes v1.NodeInterface, n *corev1.Nod
 	}
 
 	log.G(ctx).Debug("got node from api server")
-
+	node = oldNode.DeepCopy()
 	node.ResourceVersion = ""
 	node.Status = n.Status
 
 	ctx = addNodeAttributes(ctx, span, node)
 
-	updated, err := nodes.UpdateStatus(node)
+	// Patch the node status to merge other changes on the node.
+	updated, _, err := nodeutil.PatchNodeStatus(c, types.NodeName(n.Name), oldNode, node)
 	if err != nil {
 		return nil, err
 	}
 
-	log.G(ctx).WithField("resourceVersion", updated.ResourceVersion).Debug("updated node status in api server")
+	log.G(ctx).WithField("Conditions", updated.Status.Conditions).
```

Comment:
let's call this field `node.Status.Conditions`, and maybe also keep `resourceVersion` (but call it `node.resourceVersion`